pkg/cmd/render/render.go

```diff
@@ -109,98 +134,218 @@ func runRenderCmd(cmd *cobra.Command, args []string) {
+	}
+	logger := log.New()
+	logger.SetLevel(log.GetLevel())
+	disabled, err := utils.CCODisabledCheck(cm, logger)
+	if err != nil {
+		return utils.OperatorDisabledDefault
+	}
+
+	return disabled
+}
+
+type basicInstallConfig struct {
+	CredentialsMode string `yaml:"credentialsMode"`
+}
+
+func getModeFromInstallConfig() (string, error) {
+
+	// if we were not provided a place to search for the install-time manifests,
+	// just return the default cloudCredentialsMode (empty string)
+	if renderOpts.manifestsDir == "" {
+		return "", nil
+	}
+
+	cm, err := getConfigMap(renderOpts.manifestsDir, installConfigNamespace, installConfigName)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to find configmap %s/%s in manifests", installConfigNamespace, installConfigName)
+	}
+	if cm == nil {
+		return "", fmt.Errorf("failed to find configmap %s/%s in manifests", installConfigNamespace, installConfigName)
+	}
+
+	data, ok := cm.Data[installConfigKeyName]
+	if !ok {
+		return "", fmt.Errorf("did not find key %s in configmap %s/%s", installConfigKeyName, installConfigNamespace, installConfigName)
+	}
+
+	decoder := yaml.NewYAMLOrJSONDecoder(bytes.NewReader([]byte(data)), 4096)
+	instConf := &basicInstallConfig{}
+	if err := decoder.Decode(instConf); err != nil {
+		return "", errors.Wrap(err, "failed to decode install config")
+	}
+	log.Debugf("install-config contains CredentialsMode: %s", instConf.CredentialsMode)
+	return instConf.CredentialsMode, nil
+}
+
+func getConfigMap(manifestsDir, namespace, name string) (*corev1.ConfigMap, error) {
+
+	files, err := ioutil.ReadDir(manifestsDir)
+	if err != nil {
+		log.WithError(err).Errorf("failed to list files in %s", manifestsDir)
+		return nil, err
+	}
 
 	for _, fInfo := range files {
-		// non-recursive checking of all files where the the source manifests are located
+		// non-recursive checking of all files whewre the source manifests are located
 		if fInfo.IsDir() {
 			continue
 		}
 
-		fullPath := filepath.Join(renderOpts.manifestsDir, fInfo.Name())
-		log.Debugf("checking file: %s", fullPath)
+		fullPath := filepath.Join(manifestsDir, fInfo.Name())
+		log.Debugf("checking fiile: %s", fullPath)
 		file, err := os.Open(fullPath)
 		if err != nil {
 			log.WithError(err).Warn("failed to open file while searching for configmap")
 			continue
 		}
+
 		decoder := yaml.NewYAMLOrJSONDecoder(file, 4096)
 		configMap := &corev1.ConfigMap{}
 		if err := decoder.Decode(configMap); err != nil {
 			log.WithError(err).Debug("failed to decode into configmap")
 			continue
 		}
 
-		if configMap.Namespace == minterv1.CloudCredOperatorNamespace && configMap.Name == constants.CloudCredOperatorConfigMap {
-			logger := log.New()
-			logger.SetLevel(log.GetLevel())
-			disabled, err := utils.CCODisabledCheck(configMap, logger)
-			if err != nil {
-				return utils.OperatorDisabledDefault
-			}
-
-			return disabled
+		if configMap.Namespace == namespace && configMap.Name == name {
```

Comment:
This will end up matching things that are not ConfigMaps. There needs to be a check of the APIVersion and Kind as well.